src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OrganisationsController.cs

```diff
@@ -114,22 +118,36 @@ public async Task<ActionResult> UpdateOrganisationByIdAsync(Guid id, UpdateOrgan
         public async Task<ActionResult<CreateOrganisationResponseViewModel>> CreateOrganisationAsync(CreateOrganisationRequestViewModel viewModel)
         {
             if (viewModel is null)
-            {
                 throw new ArgumentNullException(nameof(viewModel));
-            }
 
-            //TODO: Task #6168
+            var result = await _createOrganisationService.CreateAsync(new CreateOrganisationRequest(
+                viewModel.OrganisationName,
+                viewModel.OdsCode,
+                viewModel.PrimaryRoleId,
+                viewModel.CatalogueAgreementSigned,
+                viewModel.Address is null ? null : new Address()
+                {
+                    Line1 = viewModel.Address.Line1,
+                    Line2 = viewModel.Address.Line2,
+                    Line3 = viewModel.Address.Line3,
+                    Line4 = viewModel.Address.Line4,
+                    Town = viewModel.Address.Town,
+                    County = viewModel.Address.County,
+                    Postcode = viewModel.Address.Postcode,
+                    Country = viewModel.Address.Country
+                }
+            ));
 
-            // Canned data
-            var firstOrganisation = (await _organisationRepository.ListOrganisationsAsync()).First();
+            var response = new CreateOrganisationResponseViewModel();
```

Comment:
There are no changes necessary now but I'll be glad when we fix the naming of these `ViewModel` classes that aren't view models. I had to do a double-take because I thought we were MVC here and returning a view but we're not – it's API.

Reply:
Agreed